pkg/gcp/secret.go

```diff
@@ -0,0 +1,92 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	"cloud.google.com/go/secretmanager/apiv1/secretmanagerpb"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+const secretFormat = "projects/%s/secrets/%s"
+
+func makeSecretURL(projectID, secretName string) string {
```

Comment:
This function returns is the name of the secret resource.
So function name `makeSecretName` is more appropriate.
https://cloud.google.com/apis/design/resource_names#resource_name_vs_url

```suggestion
func makeSecretName(projectID, secretID string) string {
```